### PR TITLE
implemented QRDecomposition[]

### PR DESCRIPTION
**Pull request by Piruzzolo**

### CODE REVIEW

File: mathics/builtin/linalg.py [hunk 1]
```diff
@@ -243,6 +244,26 @@ def apply(self, m, evaluation):
...
+    <dt>'QRDecomposition[$m$]'
+        <dd>computes the QR decomposition of the matrix $m$.
+    </dl>
+
+    >> QRDecomposition[{{1, 2}, {3, 4}, {5, 6}}]
+     = {{{Sqrt[35] / 35, 3 Sqrt[35] / 35, Sqrt[35] / 7}, {13 Sqrt[210] / 210, 2 Sqrt[210] / 105, -Sqrt[210] / 42}}, {{Sqrt[35], 44 Sqrt[35] / 35}, {0, 2 Sqrt[210] / 35}}}
+    """
+    
+    def apply(self, m, evaluation):
+        'QRDecomposition[m_]'
```
Comment (sn6uv) [inline mathics/builtin/linalg.py]: What happens if we give something that's not a matrix? e.g. QRDecomposition[4x + y]

Comment (Piruzzolo) [inline mathics/builtin/linalg.py]: It breaks. That's actually the same behavior of SingularValueDecomposition on the same input. I'm fixing it with m_?MatrixQ

**Comment by sn6uv** [comment]
The linear algebra changes look good. I'll test this out but I think this looks good to merge.

It looks like your StringReverse changes got in here as well somehow. The best way to fix this is probably with a git-rebase to remove those commits.

**Comment by Piruzzolo** [comment]
Ok, I'm going to make these fixes. Currently, I'm using the github Gui client, but I'm planning to learn deep command line git command. It will take a couple of days or more, I think. Meanwhile, feel free to merge my pull! ;)

**Comment by sn6uv** [comment]
Opened #393

**Pull request closed without merging**